main.go

```diff
@@ -244,6 +249,43 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "Mirroring")
 		os.Exit(1)
 	}
+
+	// shared event channel and sync map and scrapeInterval between the scraper and the reconciler
+	eventCh := make(chan event.GenericEvent)
+	syncMap := &sync.Map{}
+	scrapeInterval := 10 * time.Minute
+
+	if err = mgr.Add(
+		manager.RunnableFunc(func(ctx context.Context) error {
+			// run a go routine to scrape the metrics from prometheus periodically
+			scraper := (&storageautoscaler.StorageAutoscalerScraper{
+				Client:            mgr.GetClient(),
+				Log:               ctrl.Log.WithName("scraper").WithName("StorageAutoscalerScraper"),
+				OperatorNamespace: operatorNamespace,
+				ScrapeInterval:    scrapeInterval,
+				SyncMap:           syncMap,
+				EventCh:           eventCh,
+			})
+			go scraper.ScrapeMetricsPeriodically(ctx)
```

Comment:
I don't think you need to start another routine here, infact might be wrong as RunnableFunc should just [block](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager#RunnableFunc) till the end.

iirc, runtime will start a go routine with the func you supplied by passing the context.

Reply:
Maybe right, but as the current flow works I would like to keep it like that, we can do further refactoring during bug phase, @umangachapagain any quick thoughs

Reply:
not good, you can see from review there was no go routine https://github.com/red-hat-storage/ocs-operator/pull/3071#discussion_r1990710825, better not to introduce the bug knowingly.

Reply:
No, the function inside manager.RunnableFunc does not run concurrently by default. It runs once and immediately returns, meaning scrapeMetricsPeriodically() will execute only once and exit.

Reply:
>RunnableFunc implements Runnable using a function. It's very important that the given function block until it's done running.

If you understand it correctly, it says it blocks it until it is completed, if it is completed it wont re-queue.

Reply:
did you try w/ & w/o it? basically, taking a quick look you probably end up here https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/manager/runnable_group.go#L209-L229 `runnable` supplied here is the function you defined here.

an extra go routine might is not the issue I'm worried if there'll be any repercussions.

Reply:
I have tested  with `go func`, and things are running as expected. 

If I understand the `runnable code`, its just running the function provided in the background

Reply:
This would be the exact function https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/manager/runnable_group.go#L235

Reply:
trying to test it

Reply:
yaa it works fine without calling go explicitly,

Manily I was calling `go go scrape()`, understood and removed it.

thanks for pointing it out @leelavg